programs/local/LocalServer.cpp

```diff
@@ -300,7 +415,61 @@ try
         attachSystemTables(global_context);
     }
 
-    processQueries();
+    auto * history_file_from_env = getenv("CLICKHOUSE_HISTORY_FILE");
+    if (history_file_from_env)
+        history_file = history_file_from_env;
+
+    if (!history_file.empty() && !Poco::File(history_file).exists())
+        Poco::File(history_file).createFile();
+
+    LineReader::Patterns query_extenders = {"\\"};
+    LineReader::Patterns query_delimiters = {";", "\\G"};
+
+    Strings keys;
+
+    prompt_by_server_display_name = config().getRawString("prompt_by_server_display_name.default", "{display_name} :) ");
+
+    server_display_name = config().getString("display_name", getFQDNOrHostName());
+
+    /// Prompt may contain the following substitutions in a form of {name}.
+    std::map<String, String> prompt_substitutions{
+        {"display_name", server_display_name}
+    };
+
+    /// Quite suboptimal.
+    for (const auto & [key, value] : prompt_substitutions)
+        boost::replace_all(prompt_by_server_display_name, "{" + key + "}", value);
+
+
+#if USE_REPLXX
+    replxx::Replxx::highlighter_callback_t highlight_callback{};
+            if (config().getBool("highlight"))
+                highlight_callback = highlight;
+
+            ReplxxLineReader lr(*suggest, history_file, config().has("multiline"), query_extenders, query_delimiters, highlight_callback);
```

Comment:
Right now it will fail w/o `history_file`:

```
openat(AT_FDCWD, ".lock", O_RDWR|O_CREAT, 0600) = 9                                                                                                                                                                                                                               
fcntl(9, F_SETLKW, {l_type=F_WRLCK, l_whence=SEEK_CUR, l_start=0, l_len=0}) = 0                                                          
openat(AT_FDCWD, "", O_RDONLY)          = -1 ENOENT (No such file or directory)                                                          
openat(AT_FDCWD, "", O_WRONLY|O_CREAT|O_TRUNC, 0666) = -1 ENOENT (No such file or directory)                                                                                                                                                                                      
fcntl(9, F_SETLK, {l_type=F_UNLCK, l_whence=SEEK_CUR, l_start=0, l_len=0}) = 0                                                                                                                                                                                                    
close(9)                                = 0                                                                                                                                                                                                                                       
unlink(".lock")                         = 0                                                                                              
write(1, "Saving history failed: errno: 2,"..., 69Saving history failed: errno: 2, strerror: No such file or directory                                                                                                                                                            
) = 69
```